mqtt-connect/websocket/src/main/java/com/cumulocity/mqtt/connect/client/websocket/MqttWebSocketPublisher.java

```diff
@@ -1,62 +1,57 @@
 package com.cumulocity.mqtt.connect.client.websocket;
 
-import com.cumulocity.mqtt.connect.client.GenericMqttPublisher;
-import com.cumulocity.mqtt.connect.client.converter.GenericMqttMessageConverter;
-import com.cumulocity.mqtt.connect.client.exception.GenericMqttClientException;
-import com.cumulocity.mqtt.connect.client.model.GenericMqttMessage;
+import com.cumulocity.mqtt.connect.client.MqttClientException;
+import com.cumulocity.mqtt.connect.client.MqttPublisher;
+import com.cumulocity.mqtt.connect.client.model.MqttMessage;
 import com.cumulocity.sdk.client.messaging.notifications.TokenApi;
 import org.apache.commons.codec.binary.Base64;
 
 import java.net.URI;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
 
-class GenericMqttWebSocketPublisher implements GenericMqttPublisher {
-    private final static String SUBSCRIBER = "javaGenericMqttWSClientPub";
+class MqttWebSocketPublisher implements MqttPublisher {
+
+    private final static String SUBSCRIBER_NAME = "mqttConnectPublisher";
     private final static String WEBSOCKET_URL_PATTERN = "%s/notification2/producer/?token=%s";
     private final String webSocketBaseUrl;
 
-    private final GenericMqttMessageConverter genericMqttMessageConverter = new GenericMqttMessageConverter();
-
     private final AtomicInteger sequence = new AtomicInteger();
-    private final GenericMqttWebSocketConfig config;
+    private final MqttWebSocketConfig config;
     private final TokenSupplier tokenSupplier;
 
-    private GenericMqttWebSocketClient producer;
-
+    private MqttWebSocketClient producer;
 
-    GenericMqttWebSocketPublisher(String webSocketBaseUrl, TokenApi tokenApi, GenericMqttWebSocketConfig config) {
+    MqttWebSocketPublisher(String webSocketBaseUrl, TokenApi tokenApi, MqttWebSocketConfig config) {
         this.webSocketBaseUrl = webSocketBaseUrl;
         this.config = config;
-        this.tokenSupplier = new TokenSupplier(tokenApi, config.getTopic(), SUBSCRIBER);
+        this.tokenSupplier = new TokenSupplier(tokenApi, config.getTopic(), SUBSCRIBER_NAME);
```

Comment:
question: I might be missing something here, but do all subscribers have the same subscriber name? I would think that would prevent multiple clients subscribing to same topic (for subscribers, maybe not publishers as here). Unless the TokenSupplier adds some random text

Reply:
Currently supplier adds random string to the subscriber name (`representation.setSubscriber(subscriber + randomAlphabetic(10))`) which is also very bad as we will end up with multiple Pulsar subscriptions not been used and just blocking messages like it was in A1 crisis.
That's really not in scope of this PR so I will just create a follow up improvement for this.

Reply:
Ok

Reply:
https://cumulocity.atlassian.net/browse/MTM-53432